src/you_get/extractors/fc2video.py

```diff
@@ -0,0 +1,61 @@
+#!/usr/bin/env python
+
+__all__ = ['fc2video_download']
+
+from ..common import *
+from hashlib import md5
+from urllib.parse import urlparse
+import re
+
+#----------------------------------------------------------------------
+def makeMimi(upid):
+    """From http://cdn37.atwikiimg.com/sitescript/pub/dksitescript/FC2.site.js
+    Also com.hps.util.fc2.FC2EncrptUtil.makeMimiLocal
+    L110"""
+    strSeed = "gGddgPfeaf_gzyr"
+    prehash = upid + "_" + strSeed
+    return md5(prehash.encode('utf-8')).hexdigest()
+
+#----------------------------------------------------------------------
+def fc2video_download_by_upid(upid, output_dir = '.', merge = True, info_only = False, **kwargs):
+    """"""
+    fake_headers = {
+        'DNT': '1',
+        'Accept-Encoding': 'gzip, deflate, sdch',
+        'Accept-Language': 'en-CA,en;q=0.8,en-US;q=0.6,zh-CN;q=0.4,zh;q=0.2',
+        'User-Agent': 'Mozilla/5.0 (Macintosh; Intel Mac OS X 10_11_1) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/47.0.2526.58 Safari/537.36',
+        'Accept': '*/*',
+        'X-Requested-With': 'ShockwaveFlash/19.0.0.245',
+        'Connection': 'keep-alive',
+    }
+    api_base = 'http://video.fc2.com/ginfo.php?upid={upid}&mimi={mimi}'.format(upid = upid, mimi = makeMimi(upid))
+    html = get_content(api_base, headers=fake_headers)
+    
+    video_url = match1(html, r'filepath=(.+)&sec')
+    video_url = video_url.replace('&mid', '?mid')
+    
+    title = match1(html, r'&title=(.+)&seek_image')
+    
+    type, ext, size = url_info(video_url, headers=fake_headers)
```

Comment:
It turns out that I have to use this header to make it work: the built-in one won't work.

So I really hate this PR for this involves so many changes. Making things hard to test.

Should I open an issue to talk through about this major change of ```common.py```?